### ISSUE TITLE
Browser profile directories (Chrome and Firefox) are not deleted after web-ext disconnects

### ISSUE THREAD

**Issue by darkred** [mozilla/web-ext#2205]
### Is this a feature request or a bug?

Bug

Related: #1981

### What is the current behavior?
I use web-ext to build and test an extension, Refined GitHub. It located in `C:\refined-github`

Inside the above folder (cd `C:\refined-github`), 
I use `npm run start` (`web-ext run --target=chromium`) which launches Chrome stable/canary 
or `npm run start:firefox` which launches Firefox.

While the Chrome/Firefox browser instance runs, 
I've noticed that a `tmp-web-ext`/`mozilla-temp-files` folder is created in `%temp%` .

Additionally, a new temporary profile in created in :
- Chrome stable: `C:\refined-github\test\web-ext-profile\Default`
- Chrome Canary: `C:\Users\Kostas\AppData\Local\Google\Chrome SxS\Application\test`,
- Firefox: ` C:\refined-github\test\web-ext-profile`

i.e. the 1st and 3rd are created inside the current directory `C:\refined-github`, 
while the 2nd is created inside Canary installation path.

Now, when web-ext disconnects (either because I press Ctrl+C in console, or just close the browser window),
then the `tmp-web-ext`/`mozilla-temp-files` folder in `%temp%` is indeed removed,.
but the profile folder (mentioned above) and its contents all remain,
and if I re-launch `npm run start` it re-uses the profile from before.

### What is the expected or desired behavior?**

Is this the expected behavior? 
Shouldn't these new temporary profile folders be removed when web-ext disconnects?
I'd expect the temporary profile to be removed on disconnect. This would justify its "temporary" name, right?

Especially about the Canary temporary profile: it's created inside the Canary installation folder (see path above),
but it's not listed in Canary's 'Manage profiles' list, so the user has to remove it manually.

### Version information (for bug reports)

* **Chrome version**: 89 stable, 91 Canary
* **Firefox version**: 86
* **Your OS and version**: Windows 10 x64
* Paste the output of these commands: Node: v15.11.0,  npm 7.6.0, web-ext 6.0.0

````
node --version && npm --version && web-ext --version
````

**Comment by Rob--W** [comment]
`web-ext run` does not create directories in your directory by default. The refined-github project that you're referring to has configured `web-ext` to persist changes, via the `"keepProfileChanges": true` option at https://github.com/sindresorhus/refined-github/blob/ef783da57e70e82295954cfe932791a4b62f159f/package.json#L134-L136

```json
...
	"webExt": {
		"sourceDir": "distribution",
		"run": {
			"keepProfileChanges": true,
			"firefoxProfile": "./test/web-ext-profile",
			"chromiumProfile": "./test/web-ext-profile",
			"startUrl": [
				"github.com"
			]
		}
	}
}
```

Can you run `web-ext` with the `--verbose` flag, so that we can see the exact command that started Chrome?
And also the content of `chrome://version` , particularly "Command Line", "Executable Path" and "Profile Path"

**Comment by darkred** [comment]
Here:

<details>
<summary><code>web-ext</code> with the <code>--verbose</code> flag</summary>

```

C:\refined-github>web-ext run --target=chromium --verbose

[program.js][info] Version: 6.0.0
[program.js][debug] Discovering config files. Set --no-config-discovery to disable
[config.js][debug] Discovered config "C:\Users\Kostas\.web-ext-config.js" does not exist or is not readable
[config.js][debug] Discovered config "C:\refined-github\web-ext-config.js" does not exist or is not readable
[program.js][info] Applying config file: .\package.json
[config.js][debug] Loading JS config file: "C:\refined-github\package.json" (resolved to "C:\refined-github\package.json")
[config.js][debug] Looking for webExt key inside package.json file
[config.js][debug] Calling coerce() on configured value for sourceDir
[cmd/run.js][info] Running web extension from C:\refined-github\distribution
[util/manifest.js][debug] Validating manifest at C:\refined-github\distribution\manifest.json
[util/temp-dir.js][debug] Created temporary directory: C:\Users\Kostas\AppData\Local\Temp\tmp-web-ext--7832-7hsYh713yShB[extension-runners/chromium.js][debug] Creating reload-manager-extension in C:\Users\Kostas\AppData\Local\Temp\tmp-web-ext--7832-7hsYh713yShB\reload-manager-extension-1616004673515
[extension-runners/chromium.js][debug] Starting Chromium instance...
[cmd/run.js][info] The extension will reload if any source file changes
[util/file-filter.js][debug] Resolved path **/*.xpi with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\*.xpi
[util/file-filter.js][debug] Resolved path **/*.zip with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\*.zip
[util/file-filter.js][debug] Resolved path **/.* with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\.*
[util/file-filter.js][debug] Resolved path **/.*/**/* with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\.*\**\*
[util/file-filter.js][debug] Resolved path **/node_modules with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\node_modules
[util/file-filter.js][debug] Resolved path **/node_modules/**/* with sourceDir C:\refined-github\distribution to C:\refined-github\distribution\**\node_modules\**\*
[watcher.js][debug] Watching C:\refined-github\distribution for changes
[extension-runners/index.js][info] Press R to reload (and Ctrl-C to quit)

```
</details>

And 

<details>
<summary><code>chrome://version</code></summary>

```
Google Chrome	91.0.4449.0 (Official Build) canary (64-bit) (cohort: Clang-64)
Revision		0034c5812d1a49d952fca7f53d2eca76a31abe17-refs/branch-heads/4449@{#1}
OS				Windows 10 OS Version 2009 (Build 19042.867)
JavaScript		V8 9.1.140
User Agent		Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4449.0 Safari/537.36
Command Line	"C:\Users\Kostas\AppData\Local\Google\Chrome SxS\Application\chrome.exe" --remote-debugging-port=54883 --user-data-dir=./test/web-ext-profile --disable-features=TranslateUI --disable-component-extensions-with-background-pages --disable-background-networking --disable-sync --metrics-recording-only --disable-default-apps --no-default-browser-check --no-first-run --disable-backgrounding-occluded-windows --disable-renderer-backgrounding --disable-background-timer-throttling --force-fieldtrials=*BackgroundTracing/default/ --load-extension="C:\Users\Kostas\AppData\Local\Temp\tmp-web-ext--7832-7hsYh713yShB\reload-manager-extension-1616004673515,C:\refined-github\distribution" --flag-switches-begin --flag-switches-end --origin-trial-disabled-features=SecurePaymentConfirmation github.com
Executable Path	C:\Users\Kostas\AppData\Local\Google\Chrome SxS\Application\chrome.exe
Profile Path	C:\Users\Kostas\AppData\Local\Google\Chrome SxS\Application\test\web-ext-profile\Default
Variations		da89714-4ad60575   5e3a236d-59e286d0
```
i.e.
![2021-03-17_201538](https://user-images.githubusercontent.com/723651/111517267-8e0f2e80-875d-11eb-8f39-3db18b85a921.jpg)
</details>

**Comment by Rob--W** [comment]
It looks like the `--user-data-dir=./test/web-ext-profile` path is resolved relative to the working directory of Chrome. On Linux, the correct directory is selected, but on Windows, it seems that the path is resolved relative to the application.

@darkred If you change `./test/web-ext-profile` to `.\\test\\web-ext-profile` in package.json, do you see any different results?

To resolve this bug, I guess that we can canonicalize the relative path (turn it into an absolute path before use).

It is a bad idea to use a subdirectory of the current directory as the profile directory though... The extension will probably reload repeatedly, because changes are detected in the current directory.

**Comment by darkred** [comment]
> @darkred If you change `./test/web-ext-profile` to `.\\test\\web-ext-profile` in package.json, do you see any different results?

No, unfortunately, the `web-ext-profile` directory remains after disconnect.

**Comment by darkred** [comment]
> @darkred If you change `./test/web-ext-profile` to `.\\test\\web-ext-profile` in package.json, do you see any different results?

Sorry, correction: 
the directory is now created inside %temp%,
in fact the directories created are two: of which the one being nearly empty (contains 2 files, 1 KB):
<details>
<summary>screenshot</summary>

![2021-03-18_001540](https://user-images.githubusercontent.com/723651/111545913-86608180-877f-11eb-8134-8b5d1515f2e9.jpg)

</details>

and after disconnect, the small directory gets removed, but the large one (lots of files, 85 MB) remains after disconnect (even though I changed `keepProfileChanges` to false in package.json):

<details>
<summary>screenshot</summary>

![2021-03-18_001912](https://user-images.githubusercontent.com/723651/111545963-98422480-877f-11eb-8988-9fd4e6508dba.jpg)
</details>

My modified `package.json` section :

![2021-03-18_002154](https://user-images.githubusercontent.com/723651/111546232-ff5fd900-877f-11eb-9b16-539792b871e4.jpg)

**Comment by darkred** [comment]
@Rob--W Based on my previous comment, shouldn't the large folder in %temp% also be deleted after web-ext disconnects?
(I changed keepProfileChanges to `false` in package.json)

**Comment by Rob--W** [comment]
> @Rob--W Based on my previous comment, shouldn't the large folder in %temp% also be deleted after web-ext disconnects?
> (I changed keepProfileChanges to `false` in package.json)

That may be part of #1981.
